src/main/java/net/mcreator/generator/GeneratorTokens.java

```diff
@@ -58,27 +58,36 @@ public static String replaceTokens(Workspace workspace, WorkspaceSettings worksp
 	private static final Pattern brackets = Pattern.compile("@\\[(.*?)]");
 
 	static String replaceVariableTokens(Object element, String rawname) {
+		return replaceVariableTokens(element, null, rawname);
+	}
+
+	static String replaceVariableTokens(Object element, Object listItem, String rawname) {
 		if (containsVariableTokens(rawname)) {
 			Matcher m = brackets.matcher(rawname);
 			while (m.find()) {
 				String match = m.group(1);
 				Object value = null;
-				if (element != null) {
-					if (match.contains("()")) {
+				if (match.startsWith("element.")) { // a value on list item is requested

```

Comment:
I would rename this to `item.`, as one gets item from item list, not element from item list

Reply:
Maybe `listitem`?

Reply:
It is inside forEach statement so it is clear it is item for list items. Shorter name also makes more clear yaml files imo

Reply:
We also use element for ME, but item is not really used so far

Reply:
Does this apply to `elementindex` and such in templates too?

Reply:
Yep, that should be `itemindex` I would say

Reply:
✔️ Fixed

Reply:
You forgot one `@elementindex`, maybe use find all tool to make sure all are renamed